### DISCUSSION TITLE
Back up and Recovery

### DISCUSSION THREAD

**Discussion by tomwhite101** [mcguirepr89/BirdNET-Pi#752] [Q&A]
Hi All,
One of my BirdNET-Pis has gone unreachable after an update (white screen).

I think the only thing I can do is a fresh install.

So I have general questions for when this happens.

Is it possible now I have no contact with the SSL access only to recover the database file and/or the recordings?

Looking beyond this issue what is the best approach to backing up - is there any automated way to pull the database off say every 24h?

Many thanks,
Tom

**Comment by ehpersonal38** [comment]
If you can SSH into the Raspberry Pi then you'll be able to recover any files or recordings you want. If you can't SSH in then you can recover them from the SD card. 

In the future you can use rclone to backup your birds.db, you can find a tutorial online. I back mine up to my Google Drive on a weekly basis.

Once you have rclone set up and configured this is the script that will actually do the backups, you'll have to change a few parameters to match your config but it should work well:

```
#!/bin/bash

cp /home/pi/BirdNET-Pi/scripts/birds.db /home/pi/BirdNET-Pi/scripts/birds_tmp.db
/usr/bin/rclone copy --update --verbose --transfers 30 --checkers 8 --contimeout 60s --timeout 300s --retries 3 --low-level-retries 10 --min-size 10M --stats 1s "/home/pi/BirdNET-Pi/scripts/birds_tmp.db" "google-drive:PiBackups"
```

You can save that script to a file then add to your crontab to run that script however often you want. Make sure to use absolute paths in your crontab, e.g. `/usr/bin/bash /home/pi/BirdNET-Pi/backup.sh` and NOT `bash /home/pi/BirdNET-Pi/backup.sh`

**Reply by tomwhite101** [reply]
Thank you! Could you explain how to recover from SSD?

**Comment by norfolk-fr** [comment]
I wrote a small script for that : 
```
#!/bin/bash

_date="$(date +'%F')"

# Delete backups older than 7 days
find /home/pi/backup/ -type f -mtime +7 -delete
# Database Dump
sqlite3  /home/pi/BirdNET-Pi/scripts/birds.db .dump > /home/pi/backup/birds${_date}.sql
# Compress the dump
gzip -9 --rsyncable -f /home/pi/backup/birds${_date}.sql
# Send the dump to another raspberry for storage
rsync -avrq /home/pi/backup/* pi@toolbox:/home/pi/backup_birdnet/
```
To restore the database, a simple `sqlite3 birds.db < dumpfile` will do the job.

Hope this helps.
Alban
